qiskit/algorithms/eigen_solvers/vqd.py

```diff
@@ -0,0 +1,840 @@
+                for the optimizer. If ``None`` then VQE will look to the ansatz for a preferred
+                point and if not will simply compute a random one.
+            gradient: An optional gradient function or operator for optimizer.
+                Only used to compute the ground state at the moment.
+            expectation: The Expectation converter for taking the average value of the
+                Observable over the ansatz state function. When ``None`` (the default) an
+                :class:`~qiskit.opflow.expectations.ExpectationFactory` is used to select
+                an appropriate expectation based on the operator and backend. When using Aer
+                qasm_simulator backend, with paulis, it is however much faster to leverage custom
+                Aer function for the computation but, although VQE performs much faster
+                with it, the outcome is ideal, with no shot noise, like using a state vector
+                simulator. If you are just looking for the quickest performance when choosing Aer
+                qasm_simulator and the lack of shot noise is not an issue then set `include_custom`
+                parameter here to ``True`` (defaults to ``False``).
+            include_custom: When `expectation` parameter here is None setting this to ``True`` will
+                allow the factory to include the custom Aer pauli expectation.
+            max_evals_grouped: Max number of evaluations performed simultaneously. Signals the
+                given optimizer that more than one set of parameters can be supplied so that
+                multiple points to compute the gradient can be passed and if computed in parallel
+                potentially the expectation values can be computed in parallel. Typically this is
+                possible when a finite difference gradient is used by the optimizer such that
+                improve overall execution time. Deprecated if a gradient operator or function is
+                given.
+            callback: a callback that can access the intermediate data during the optimization.
+                Four parameter values are passed to the callback as follows during each evaluation
+                by the optimizer for its current set of parameters as it works towards the minimum.
+                These are: the evaluation count, the optimizer parameters for the
+                ansatz, the evaluated mean and the evaluated standard deviation.`
+            quantum_instance: Quantum Instance or Backend
+
+        """
+        validate_min("max_evals_grouped", max_evals_grouped, 1)
+
+        super().__init__()
+
+        self._max_evals_grouped = max_evals_grouped
+        self._circuit_sampler = None  # type: Optional[CircuitSampler]
+        self._expectation = None
+        self.expectation = expectation
+        self._include_custom = include_custom
+
+        # set ansatz -- still supporting pre 0.18.0 sorting
```

Comment:
You marked the conversation for ansatz and sorting resolved - should we not get rid of the comment here too?